test/boost/secondary_index_test.cc

```diff
@@ -1208,6 +1208,279 @@ SEASTAR_TEST_CASE(test_indexing_paging_and_aggregation) {
     });
 }
 
+// Verifies that both "SELECT * [rest_of_query]" and "SELECT count(*) [rest_of_query]" 
+// return expected count of rows.
+void assert_select_count_and_select_rows_has_size(cql_test_env& e, const sstring& rest_of_query, int64_t expected_count) {
```

Comment:
Making this `void` will diminish the value of diagnostic messages in future failures.  Consider returning results from this function and asserting on them in the caller OR passing a `source_location` parameter like `require_rows` does.